src/fields/core/fieldInput.vue

```diff
@@ -38,168 +39,167 @@
+        this.updateModelValue(newValue, oldValue);
+      },
+      onInput($event) {
+        let value = $event.target.value;
+        switch (this.schema.inputType.toLowerCase()) {
+          case "number":
+          case "range":
+            if (isNumber(parseFloat($event.target.value))) {
+              value = parseFloat($event.target.value);
+            }
+            break;
+        }
+        this.value = value;
+      },
+    },
 
-			return value;
-		},
-		formatValueToField(value) {
-			switch(this.schema.inputType.toLowerCase()) {
-				case "date":
-				case "datetime":
-				case "datetime-local":
-					return this.formatDatetimeValueToField(value);
-			}
-			return value;
-		},
-		formatDatetimeToModel(newValue, oldValue) {
-			let defaultFormat = DATETIME_FORMATS[this.schema.inputType.toLowerCase()];
-			let m = fecha.parse(newValue, defaultFormat);
-			if (m !== false) {
-				if (this.schema.format) {
-					newValue = fecha.format(m, this.schema.format);
-				} else {
-					newValue = m.valueOf();
-				}
-			}
-			this.updateModelValue(newValue, oldValue);
-		},
-		formatDatetimeValueToField(value) {
-			if(value === null || undefined === value) {
-				return null;
-			}
+    mounted() {
+      switch (this.schema.inputType.toLowerCase()) {
+        case "number":
+        case "range":
+          this.debouncedFormatFunc = debounce(
+            (newValue, oldValue) => {
+              this.formatNumberToModel(newValue, oldValue);
+            },
+            parseInt(objGet(this.schema, "debounceFormatTimeout", 1000)),
+            {
+              trailing: true,
+              leading: false
+            }
+          );
+          break;
+        case "date":
+        case "datetime":
+        case "datetime-local":
+          // wait 1s before calling 'formatDatetimeToModel' to allow user to input data
+          this.debouncedFormatFunc = debounce(
+            (newValue, oldValue) => {
+              this.formatDatetimeToModel(newValue, oldValue);
+            },
+            parseInt(objGet(this.schema, "debounceFormatTimeout", 1000)),
+            {
+              trailing: true,
+              leading: false
+            }
+          );
+          break;
+      }
+    },
 
-			let defaultFormat = DATETIME_FORMATS[this.schema.inputType.toLowerCase()];
-			let m = value;
-			if(!isNumber(value)) {
-				m = fecha.parse(value, defaultFormat);
-			}
-			if(m !== false) {
-				return fecha.format(m, defaultFormat);
-			}
-			return value;
-		},
-		formatNumberToModel(newValue, oldValue) {
-			if (!isNumber(newValue)) {
-				newValue = NaN;
-			}
-			this.updateModelValue(newValue, oldValue);
-		},
-		onInput($event) {
-			let value = $event.target.value;
-			switch (this.schema.inputType.toLowerCase()) {
-				case "number":
-				case "range":
-					if (isNumber(parseFloat($event.target.value))) {
-						value = parseFloat($event.target.value);
-					}
-					break;
-			}
-			this.value = value;
-		},
-		onBlur() {
```

Comment:
It looks like you are removing functionality here. I think we should keep this `onBlur` and in this `onBlur` call `this.onDataAccept`

Reply:
Thanks for the heads up. I cleaned up! I just called the this.onDataAccept after the current functionality.